cluster-ui: tenants use sqlstats-supplied regions

matthewtodd opened the pull request:
Fixes #98056.

As of #95449, the SQL Activity pages in the DB Console can draw regions information directly from the sqlstats tables, rather than having to translate node IDs to regions on page load.
    
Here, we make that switch, but for non-system tenants only, because:
    
1. The ephemeral nature of serverless nodes made this view-time mapping especially problematic in that context. (See further notes in #95449.)
    
2. The system-tenant views also include KV node IDs in a special Regions/Nodes column, which we are unable to recreate given the backend storage structure. (Future design work might suggest removing these node IDs altogether, for a unified UI.)

# Screenshots!
## Statements, with and without regions filter
<img width="1372" alt="statements" src="https://user-images.githubusercontent.com/5261/225033247-739df90a-9173-4aab-a666-a61a1ceeb579.png">
<img width="1372" alt="statements - filtered" src="https://user-images.githubusercontent.com/5261/225033271-1c0d0f82-3dd4-48ea-bdef-11f19af97a85.png">

## Statement details
<img width="1372" alt="statement details" src="https://user-images.githubusercontent.com/5261/225033338-6dff4a6e-a4a3-48c6-863a-84f1375b0a61.png">

## Transactions, with and without regions filter
<img width="1372" alt="transactions" src="https://user-images.githubusercontent.com/5261/225033366-65f44e95-3549-47cc-b0f2-67ad48a1a1fa.png">
<img width="1372" alt="transactions - filtered" src="https://user-images.githubusercontent.com/5261/225033391-50b9a2dc-e9a1-457b-84b1-837426eba35e.png">

## Transaction details
<img width="1372" alt="transaction details" src="https://user-images.githubusercontent.com/5261/225033505-3fdeceef-35dc-4e06-af25-ab4d0c53518f.png">

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/98410)
<!-- Reviewable:end -->

Commit:
cluster-ui: tenants use sqlstats-supplied regions

Fixes #98056.

As of #95449, the SQL Activity pages in the DB Console can draw regions
information directly from the sqlstats tables, rather than having to
translate node IDs to regions on page load.

Here, we make that switch, but for non-system tenants only, because:

1. The ephemeral nature of serverless nodes made this view-time mapping
   especially problematic in that context. (See further notes in #95449.)

2. The system-tenant views also include KV node IDs in a special
   Regions/Nodes column, which we are unable to recreate given the
   backend storage structure. (Future design work might suggest removing
   these node IDs altogether, for a unified UI.)

Release note: None

matthewtodd commented:
@maryliag we'll want to coordinate when to put this into a cluster-ui release.

maryliag reviewed (commented):
> we'll want to coordinate when to put this into a cluster-ui release.

what is the concern of what should be added together?

> Reviewed 12 of 12 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/98410#-:-NQViFUQ29peno86ybSr:b-xgeuk5)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @matthewtodd)



<!-- Sent from Reviewable.io -->

matthewtodd commented:
Thanks, @maryliag!

bors r+
